### PR TITLE
LT-912 add an external note to a case

### PR DESCRIPTION
**Pull request by cstobart**

### CODE REVIEW

File: applications/urls.py [hunk 1]
```diff
@@ -7,4 +7,5 @@
 urlpatterns = [
     path('', views.ApplicationList.as_view(), name='applications'),
     path('<uuid:pk>/', views.ApplicationDetail.as_view(), name='application'),
+    path('<uuid:pk>/pk-user/', views.ApplicationDetailPkUser.as_view(), name='application-pk-user'),
```
Comment (efahy) [inline applications/urls.py]: API URLs shouldn't have `pk` in them, please change this to `'<uuid:pk>/user/'`

File: applications/views.py [hunk 2]
```diff
@@ -126,11 +128,28 @@ def put(self, request, pk):
...
-                reversion.add_meta(GovUserRevisionMeta, gov_user=request.user)
+                reversion.set_user(self.request.user)
 
                 serializer.save()
                 return JsonResponse(data={'application': serializer.data})
 
             return JsonResponse(data={'errors': serializer.errors}, status=400)
+
+
+class ApplicationDetailPkUser(ApplicationDetail):
```
Comment (efahy) [inline applications/views.py]: Rename to `ApplicationDetailUser`, in line with the suggested API URL change

File: clc_queries/views.py [hunk 3]
```diff
@@ -5,7 +5,9 @@
 from clc_queries.libraries.get_clc_query import get_clc_query_by_pk
 from clc_queries.serializers import ClcQueryUpdateSerializer
 from conf.authentication import GovAuthentication
-from gov_users.models import GovUserRevisionMeta
+
+
+# from users.models import GovUserRevisionMeta
```
Comment (efahy) [inline clc_queries/views.py]: Remove commented out code

File: clc_queries/views.py [hunk 4]
```diff
@@ -19,8 +21,8 @@ def put(self, request, pk):
             serializer = ClcQueryUpdateSerializer(get_clc_query_by_pk(pk), data=request.data, partial=True)
 
             if serializer.is_valid():
-                reversion.set_comment("Updated clc query details")
-                reversion.add_meta(GovUserRevisionMeta, gov_user=request.user)
+                # reversion.set_comment("Updated clc query details")
+                # reversion.add_meta(GovUserRevisionMeta, gov_user=request.user)
```
Comment (efahy) [inline clc_queries/views.py]: Remove if not needed

File: goods/models.py [hunk 5]
```diff
@@ -18,3 +17,8 @@ class Good(models.Model):
     organisation = models.ForeignKey(Organisation, on_delete=models.CASCADE, default=None)
     status = models.CharField(choices=GoodStatus.choices, default=GoodStatus.DRAFT, max_length=20)
     # type_of_licence = models.CharField() --with this with could split the list of goods and the open license high level description of goods
+
+    # @property
+    # def notes(self):
+    #     from cases.models import CaseNote  # circular import prevention
```
Comment (efahy) [inline goods/models.py]: Remove commented out code, assuming not needed

File: applications/urls.py (hunk 1 above)
Comment (5pence) [reply applications/urls.py]: changed as suggest

File: applications/views.py (hunk 2 above)
Comment (5pence) [reply applications/views.py]: refactored as suggested

File: clc_queries/views.py (hunk 3 above)
Comment (5pence) [reply clc_queries/views.py]: removed

File: clc_queries/views.py (hunk 4 above)
Comment (5pence) [reply clc_queries/views.py]: removed

File: goods/models.py (hunk 5 above)
Comment (5pence) [reply goods/models.py]: removed

**Pull request merged**